neuralmonkey/writers/plain_text_writer.py

```diff
@@ -1,34 +1,58 @@
 from typing import Iterator, List, Any, Callable
 
 from neuralmonkey.logging import log
+from neuralmonkey.readers.plain_text_reader import _ALNUM_CHARSET
 
 # pylint: disable=invalid-name
 # Writer: function that gets file and the data
 Writer = Callable[[str, Any], None]
 # pylint: enable=invalid-name
 
 
-def tokenized_text_writer(encoding: str = "utf-8") -> Writer:
+def text_writer(encoding: str = "utf-8") -> Writer:
 
-    def writer(path: str, data: Iterator[List[str]]) -> None:
+    def writer(path: str, data: Iterator[Any]) -> None:
         with open(path, "w", encoding=encoding) as f_out:
             for sentence in data:
-                f_out.write(" ".join(sentence) + "\n")
-        log("Result saved as tokenized text in '{}'".format(path))
+                f_out.write(str(sentence) + "\n")
+        log("Result saved as plain text in '{}'".format(path))
 
     return writer
 
 
-def text_writer(encoding: str = "utf-8") -> Writer:
+def tokenized_text_writer(encoding: str = "utf-8") -> Writer:
 
-    def writer(path: str, data: Iterator[Any]) -> None:
-        with open(path, "w", encoding=encoding) as f_out:
-            for sentence in data:
-                f_out.write(str(sentence) + "\n")
-        log("Result saved as plain text in '{}'".format(path))
+    def writer(path: str, data: Iterator[List[str]]) -> None:
+        writer = text_writer(encoding)
```

Comment:
overloaded name, use something else that don't collide with the name of the function (or rename the function)